stmt/stmts/grant.go

```diff
@@ -0,0 +1,761 @@
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
+		}
+		// Grant each priv to the user.
+		for _, priv := range s.Privs {
+			if len(priv.Cols) > 0 {
+				// Check column scope privilege entry.
+				// TODO: Check validity before insert new entry.
+				err1 := s.checkAndInitColumnPriv(ctx, userName, host, priv.Cols)
+				if err1 != nil {
+					return nil, errors.Trace(err1)
+				}
+			}
+			err2 := s.grantPriv(ctx, priv, user)
+			if err2 != nil {
+				return nil, errors.Trace(err2)
+			}
+		}
+	}
+	return nil, nil
+}
+
+// Check if DB scope privilege entry exists in mysql.DB.
+// If unexists, insert a new one.
+func (s *GrantStmt) checkAndInitDBPriv(ctx context.Context, user string, host string) error {
+	db, err := s.getTargetSchema(ctx)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	ok, err := dbUserExists(ctx, user, host, db.Name.O)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if ok {
+		return nil
+	}
+	// Entry does not exists for user-host-db. Insert a new entry.
+	return initDBPrivEntry(ctx, user, host, db.Name.O)
+}
+
+// Check if table scope privilege entry exists in mysql.Tables_priv.
+// If unexists, insert a new one.
+func (s *GrantStmt) checkAndInitTablePriv(ctx context.Context, user string, host string) error {
+	db, tbl, err := s.getTargetSchemaAndTable(ctx)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	ok, err := tableUserExists(ctx, user, host, db.Name.O, tbl.TableName().O)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if ok {
+		return nil
+	}
+	// Entry does not exists for user-host-db-tbl. Insert a new entry.
+	return initTablePrivEntry(ctx, user, host, db.Name.O, tbl.TableName().O)
+}
+
+// Check if column scope privilege entry exists in mysql.Columns_priv.
+// If unexists, insert a new one.
+func (s *GrantStmt) checkAndInitColumnPriv(ctx context.Context, user string, host string, cols []string) error {
+	db, tbl, err := s.getTargetSchemaAndTable(ctx)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	for _, c := range cols {
+		col := column.FindCol(tbl.Cols(), c)
+		if col == nil {
+			return errors.Errorf("Unknown column: %s", c)
+		}
+		ok, err := columnPrivEntryExists(ctx, user, host, db.Name.O, tbl.TableName().O, col.Name.O)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		if ok {
+			continue
+		}
+		// Entry does not exists for user-host-db-tbl-col. Insert a new entry.
+		err = initColumnPrivEntry(ctx, user, host, db.Name.O, tbl.TableName().O, col.Name.O)
+		if err != nil {
+			return errors.Trace(err)
+		}
+	}
+	return nil
+}
+
+// Insert a new row into mysql.DB with empty privilege.
+func initDBPrivEntry(ctx context.Context, user string, host string, db string) error {
+	st := &InsertIntoStmt{
+		TableIdent: table.Ident{
+			Name:   model.NewCIStr(mysql.DBTable),
+			Schema: model.NewCIStr(mysql.SystemDB),
+		},
+		ColNames: []string{"Host", "User", "DB"},
+	}
+	values := make([][]expression.Expression, 0, 1)
+	value := make([]expression.Expression, 0, 3)
```

Comment:
value := []expression.Expression{
   &expression.Value{Val: host},
   &expression.Value{Val: user},
   &expression.Value{Val: db},
}

it's neater.